packages/flutter/lib/src/material/carousel.dart

```diff
@@ -503,7 +503,7 @@ class _CarouselViewState extends State<CarouselView> {
           Axis.vertical => constraints.maxHeight,
         };
         _itemExtent =
-            _itemExtent == null ? _itemExtent : clampDouble(_itemExtent!, 0, mainAxisExtent);
+            widget.itemExtent == null ? null : clampDouble(widget.itemExtent!, 0, mainAxisExtent);
```

Comment:
Just curious, why do we want to have this change here?

Reply:
Not obvious, but in fact it is crucial to fix this issue.
Without this change, when the viewport dimension is zero at first, _itemExtent is clamped to 0. Later when the viewport gets the final dimension, the old code `clampDouble(_itemExtent!, 0, mainAxisExtent); `still returns 0.0 and never check again the original widget.itemExtent value.
This is covered by the test named "itemExtent is applied when viewport dimension is updated".

Reply:
Ah nice catch! Thanks for the explanation!